scripts/memory/__init__.py

```diff
@@ -0,0 +1,42 @@
+from memory.local import LocalCache
+try:
+    from memory.redismem import RedisMemory
+except ImportError:
+    print("Redis not installed. Skipping import.")
+    RedisMemory = None
+
+try:
+    from memory.pinecone import PineconeMemory
+except ImportError:
+    print("Pinecone not installed. Skipping import.")
+    PineconeMemory = None
+
+
+def get_memory(cfg, init=False):
+    memory = None
+    if cfg.memory_backend == "pinecone":
+        if not PineconeMemory:
+            print("Error: Pinecone is not installed. Please install pinecone"
+                  " to use Pinecone as a memory backend.")
+        else:
+            memory = PineconeMemory(cfg)
+            if init:
+                memory.clear()
+    elif cfg.memory_backend == "redis":
+        if not RedisMemory:
+            print("Error: Redis is not installed. Please install redis-py to"
+                  " use Redis as a memory backend.")
+        else:
+            memory = RedisMemory(cfg)
+
+    if memory is None:
+        memory = LocalCache(cfg)
```

Comment:
We appear to be missing a call to memory.clear() here.

Reply:
For the LocalCache

Reply:
Yeah, there was a lot of people that wanted the memory to persist by default and I just made Pi's implementation work.

I guess a direction one way or the other needs to be picked.

Deleting the memory is easy enough for local, use just delete the json file.

But there's also a memory index if you wanted to start fresh but keep memory from a different session.

Reply:
Interesting, okay.

Perhaps default persistent memory is a good idea in the near future, however now I think it has the side effect of encouraging the AI to repeat it's errors.

Through my experimenting with this PR, if the AI is reminded of events from it's past were it didn't respond in JSON, it's more likely to do so again.

Reply:
It might be prudent to not even save those in memory.